### ISSUE TITLE
ag-grid-vue3 emit event issue when use Framework Component with keep-alive, and the App created by VueComponentFactory.createAndMountComponent function is not unmount when ag-grid-vue3 unmount.

### ISSUE THREAD

**Issue by yuanjinyong** [ag-grid/ag-grid#4618]
I used ag-grid-vue3 25.3.1  in my project, and encountered the two issue.

My code like following:
=====================================================
1> Three component corresponding three router, each component have an ag-grid-vue3
```
<template>
    <ag-grid-vue  class="ag-theme-fresh" :grid-options="gridOptions" />
</template>
<script>
...... // omit some lines
import defaultGridOptions from './defaultGridOptions'

export default defineComponent({
  name: 'JyGrid1', // anothers are JyGrid2 and JyGrid3
  components: {AgGridVue},
  props: {
    detail: {type: Object, default: () => reactive({})}, // {component: () => this.$refs.detail, operations: ['edit', 'remove']}
  },
...... // omit some lines
  created() {
...... // omit some lines
    this.gridOptions.context.grid = this // in OperationRendererFramework will use context.grid to call onDetailOperation funcation
    this.gridOptions.columnDefs = this._buildColumnDefs(this.$attrs.columnDefs || [])
...... // omit some lines
  },
  methods: {
    _buildColumnDefs(columnDefs) {
      let colDefs = []
...... // omit some lines
      // 6、添加操作列
      if (this.detail.operations?.length) {
        const operations = this.$cfg.buildOperations('detail', this.name, this.detail.operations)
        const colDef = {type: ['OperationRender'], cellRendererParams: {operations}}
        colDef.width = colDef.cellRendererParams.operations.length * this.$cfg.operationWidth + 6
        colDefs.push(colDef)
      }

      return colDefs
    }
})
</script>
```

2> In defaultGridOptions file defined columnTypes is:
```
OperationRender: {
    headerName: '操作',
    pinned: 'right',
    resizable: false,
    suppressExport: true,
    cellStyle: {'text-align': 'center'},
    cellRendererFramework: 'OperationRendererFramework'
  }
```

3> And in OperationRendererFramework, v-on the operation event:
```
<template>
  <jy-operation-group mode="detail" :operations="params.operations" :permissions="params.context.grid.permissions" @operation="onOperation" />
</template>

<script>
import {defineComponent} from 'vue'

export default defineComponent({
  name: 'OperationRendererFramework',
...... // omit some lines
  created() {
    this.params.operations.forEach(operation => (operation.disabled = this.isDisabled(operation)))
  },
  methods: {
...... // omit some lines
    onOperation(operation) {
      console.log('onOperation', this.$.uid, this.params.context.grid.name, this.params.context.grid.$options.name, this.params.context.grid.$.uid, operation)
      this.params.context.grid.onDetailOperation({operation, entity: this.entity, params: this.params})
    }
  }
})
</script>
```

4> In Main component, use JyGrid1, JyGrid2 and JyGrid3 by switch router, and use keep-alive:
```
<template>
    <jy-main class="jy-admin-main__page">
      <router-view v-slot="{Component, route}">
        <keep-alive ref="keepAlive"><component class="jy-teleport" :key="route.path" :is="Component" /></keep-alive>
      </router-view>
    </jy-main>
</template>
```

The first issue:
=====================================================
1> Route to JyGrid1, JyGrid2 and JyGrid3 one by one, the last one created is JyGrid3, then switch route to JyGrid1, and click an operation button in JyGrid1 to trigger the operation event, but JyGrid3 will receive the event, not JyGrid1.
2> Route to JyGrid1, JyGrid3 and JyGrid2 one by one, the last one created is JyGrid2, then switch route to JyGrid1, and click an operation button in JyGrid1 to trigger the operation event, but JyGrid2 will receive the event, not JyGrid1.
3> If change the Main component, remove keep-alive, then all is OK.
Maybe the event bus of vue don't work in this situation? The listeners by last one created will override the listeners created by before if enable keep-alive?

The second issue:
=====================================================
When unmount all of three JyGrid, the App created by VueComponentFactory.createAndMountComponent function is not unmount(destroy), I add the log in VueFrameworkComponentWrapper.ts, it is not print the log message:
```
abstract class VueComponent<P, T> {
......
    public destroy(): void {
        console.log('VueComponent.destroy', this.mountedComponent)
        this.mountedComponent.unmount();
    }
......
}

```
And I check the App count from Vue DevTools 6.0.0 beta 15, there remain have hundreds of app.
And the memory used by browser don't release.

**Comment by yuanjinyong** [comment]
The lifecycle hooks of Vue3 beforeDestroy and destroyed have been renamed to beforeUnmount and unmounted respectively.
I change the code to following can unmount app:
```
<template>
    <ag-grid-vue ref="agGridVue" class="ag-theme-fresh" :grid-options="gridOptions" />
</template>
<script>
...... // omit some lines
import defaultGridOptions from './defaultGridOptions'

export default defineComponent({
  name: 'JyGrid1', // anothers are JyGrid2 and JyGrid3
...... // omit some lines
  beforeUnmount() {
    this.$refs.agGridVue.destroyed() // call destroyed
  },
})
</script>
```

**But the still has issue when refresh the grid, and the cell with Framework Component whill render, whill create a new app, but the old app don't unmount.**

**Comment by yuanjinyong** [comment]
I have tried change AgGridVue.ts file 
from
public destroyed() {
to
public unmounted() {

Then the app can unmount, and the memory used by browser is released.

**Comment by yuanjinyong** [comment]
The first issue is caused by "this.params.context" was wrong,

**Comment by makinggoodsoftware** [comment]
Hi guys,

v26 was released a few days ago, Vue3 support was enhanced there, can you check if this issue is still persisting in that version?

Thanks

**Comment by yuanjinyong** [comment]
@makinggoodsoftware 
When refresh the grid, can unmount Framework Component app and create a new Framework Component app, but when close the tab to unmount AgGridVue, the Framework Component app doesn't unmount.

Because of the lifecycle hooks of Vue3 beforeDestroy and destroyed have been renamed to beforeUnmount and unmounted, so the code in AgGridVue.ts file need change
from
```
public destroyed() {
```
to
```
public unmounted() {
```

**Comment by seanlandsman** [comment]
@yuanjinyong thanks for the feedback - I'll get this change done

**Comment by seanlandsman** [comment]
I've published `26.0.1-beta.0` of both `@ag-grid-community/vue3` and `ag-grid-vue3` 

Let me know if this fixes the above issue for you! 

thanks

**Comment by yuanjinyong** [comment]
There are some errors, my Framework Component code is following:
```
<script>
import {defineComponent, h} from 'vue'

export default defineComponent({
  name: 'TimestampRendererFramework',
  render() {
    return h('span', {}, this.$util.formatTimestamp(this.params.value, this.params.format))
  }
})
</script>
```

Error message is:
```
[Vue warn]: Property "params" was accessed during render but is not defined on instance. 
  at <TimestampRendererFramework params= {valueFormatted: null, node: RowNode2, getValue: ƒ, setValue: ƒ, formatValue: ƒ, …}

TypeError: Cannot read property 'value' of undefined
    at Proxy.render (TimestampRendererFramework.vue:7)
    at renderComponentRoot (runtime-core.esm-bundler.js:761)
    at ReactiveEffect.componentUpdateFn [as fn] (runtime-core.esm-bundler.js:4531)
    at ReactiveEffect.run (reactivity.esm-bundler.js:160)
    at setupRenderEffect (runtime-core.esm-bundler.js:4657)
    at mountComponent (runtime-core.esm-bundler.js:4440)
    at processComponent (runtime-core.esm-bundler.js:4398)
    at patch (runtime-core.esm-bundler.js:3993)
    at Object.render2 [as render] (runtime-core.esm-bundler.js:5142)
    at render (runtime-dom.esm-bundler.js:1489)
```

**Comment by seanlandsman** [comment]
I have just published `26.0.1-beta.1` - do you want to give that a go and let me know?

thanks

**Comment by yuanjinyong** [comment]
Great, good job.
The change from createApp to createVnode, resolved many issues, eg: HMR, performance.
